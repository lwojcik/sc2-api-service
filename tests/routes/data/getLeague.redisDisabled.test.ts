import fastify from 'fastify';
import server from '../../../src/index';
import getConfig from '../../helper';

describe('/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis disabled)', () => {
  const fastifyServer = fastify() as any;
  const url = '/data/league/40/201/0/6';

  beforeAll(() => fastifyServer.register(server, getConfig(false)));

  afterAll(() => fastifyServer.close());

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.payload).toMatchSnapshot();
  });
});
