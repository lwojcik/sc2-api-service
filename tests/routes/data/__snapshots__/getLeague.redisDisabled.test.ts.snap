// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"_links\\":{\\"self\\":{\\"href\\":\\"sample league object\\"}}}}"`;

exports[`/data/league/:seasonId/:queueId/:teamType/:leagueId (Redis disabled) returns correct response when refresh is set to true 1`] = `"{\\"status\\":200,\\"data\\":{\\"_links\\":{\\"self\\":{\\"href\\":\\"sample league object\\"}}}}"`;
